packages/blocks/src/api/raw-handling/shortcode-converter.js

```diff
@@ -36,17 +36,20 @@ function segmentHTMLToShortcodeBlock( HTML, lastIndex = 0, excludedBlockNames =
 	const previousIndex = lastIndex;
 
 	if ( ( match = next( transformTag, HTML, lastIndex ) ) ) {
-		const beforeHTML = HTML.substr( 0, match.index );
-
 		lastIndex = match.index + match.content.length;
+		const beforeHTML = HTML.substr( 0, match.index );
+		const afterHTML = HTML.substr( lastIndex );
 
 		// If the shortcode content does not contain HTML and the shortcode is
 		// not on a new line (or in paragraph from Markdown converter),
 		// consider the shortcode as inline text, and thus skip conversion for
 		// this segment.
 		if (
 			! includes( match.shortcode.content || '', '<' ) &&
-			! /(\n|<p>)\s*$/.test( beforeHTML )
+			! (
+				/(\n|<p>)\s*$/.test( beforeHTML ) &&
```

Comment:
Minor: Maybe here too?

Reply:
Actually, on second thought I'm going to roll back on the non-capturing group changes. I'm thinking that the fact that we're testing with `RegExp::test` instead of `::exec` or `String::match` should be enough — and if not, that's something for the VMs to work on!

Quoting from [`RegExp::test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test):

> Use test() whenever you want to know whether a pattern is found in a string. test() returns a boolean, unlike the String.prototype.search() method, which returns the index (or -1 if not found). To get more information **(but with slower execution)**, use the exec() method (similar to the String.prototype.match() method). As with exec() (or in combination with it), test() called multiple times on the same global regular expression instance will advance past the previous match.

Reply:
Sounds good! Thanks for looking into it! :)